### PR TITLE
Cryptography utils

### LINKED ISSUE #52: Cryptography Utils
ℹ️ This issue is part of Epic #51 

## Checklist

- [x] bin_to_hex
- [x] hex_to_bin
- [x] url_encode64
- [x] url_decode64
- [x] blake2b_hash

### PR DESCRIPTION
**Pull request by juanhurtado10**
## Description

Fixes #52 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/cryptography/utils.ex [hunk 1]
```diff
@@ -0,0 +1,29 @@
+defmodule Kadena.Cryptography.Utils do
+  @moduledoc """
+  Documentation for `Cryptography` utils functions.
```
Comment (boterop) [inline lib/cryptography/utils.ex]: IMO to have consistency in the module doc from all modules, you should specify `Cryptography.Utils`

File: lib/cryptography/utils.ex [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  """
+  @type hash :: String.t()
+  @type str :: String.t()
+  @type key :: String.t()
+  @type opts :: Keyword.t()
+
+  @spec bin_to_hex(key_bin :: binary()) :: String.t()
```
Comment (boterop) [inline lib/cryptography/utils.ex]: I think the return in this spec could be a Kadena.Type.Base16String

File: lib/cryptography/utils.ex [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  def bin_to_hex(key_bin), do: Base.encode16(key_bin, case: :lower)
+
+  @spec hex_to_bin(key :: String.t()) :: binary()
```
Comment (boterop) [inline lib/cryptography/utils.ex]: We could be more specific in Key type, for instance, Kadena.Type.Base16String

File: lib/cryptography/utils.ex [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  def hex_to_bin(key), do: Base.decode16!(key, case: :lower)
+
+  @spec url_encode64(hash_bin :: binary()) :: String.t()
```
Comment (boterop) [inline lib/cryptography/utils.ex]: IMO the return type should be a Kadena.Type.Base64Url

File: lib/cryptography/utils.ex [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  def url_encode64(hash_bin), do: Base.url_encode64(hash_bin, padding: false)
+
+  @spec url_decode64(hash :: String.t()) :: binary()
```
Comment (boterop) [inline lib/cryptography/utils.ex]: Even here in the hash type, I think we could use Base64Url type

File: lib/cryptography/utils.ex (hunk 2 above)
Comment (juanhurtado10) [reply lib/cryptography/utils.ex]: Agree!

File: lib/cryptography/utils.ex (hunk 3 above)
Comment (juanhurtado10) [reply lib/cryptography/utils.ex]: Agree!

**Review by boterop — APPROVED** [review]

**Pull request merged**